{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## CODIGO PRINCIPAL"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "En este notebook se incluyen las ejecuciones de los programas desarrollados en el proyecto de forma secuencial.\n",
    "\n",
    "En concreto este código está dedicado a los capítulos del trabajo indicados a continuación:\n",
    "\n",
    "    3. Sistema, diseño y desarrollo\n",
    "    4. Experimentos realizados y resultados"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Importar funciones y módulos requeridos\n",
    "\n",
    "El módulo *nimgenetics.py* contiene las funciones necesarias para ejecutar este código."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import sys\n",
    "\n",
    "sys.path.append('./my_module/')\n",
    "import nimgenetics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 3. SISTEMA, DISEÑO Y DESARROLLO\n",
    "\n",
    "     \n",
    "Directorios importantes:\n",
    "     \n",
    "- folder_path: carpeta donde se ubican los informes de NIMGenetics.\n",
    "- folder_result: carpeta donde se irán guardando los resultados de las ejecuciones de los programas.\n",
    "- pathI: path de la carpeta \"Identificados\" de los informes que cumplen el patrón de selección.\n",
    "- image_path: carpeta donde se almacenan las imágenes generadas a partir de los informes.\n",
    "- textos: carpeta donde se almacenan los textos generados a partir de las imágenes."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "folder_path = 'path_to_folder_root'\n",
    "folder_result = 'path_to_folder_result'\n",
    "pathI = 'path_to_folder_identificados'\n",
    "image_path = 'path_to_images'\n",
    "textos = 'path_to_textos'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 3.2 Obtención de informes clínicos de Gestlab\n",
    "Informes filtrados y renombrados."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 3.2.1 Mapeo de los informes localizados en folder_path\n",
    "La función load_files es utilizada para mapear los informes (PDF) dentro de folder_path."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# l_path contiene el path absoluto de todos los informes\n",
    "l_path = nimgenetics.load_files(root_dir=folder_path)[0]\n",
    "# l_pdf es el mapeo individual de cada informe\n",
    "l_pdf = nimgenetics.load_files(root_dir=folder_path)[1]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 3.2.2 Obtención de los identificadores reales de los informes\n",
    "La función getIDs realiza una consulta sql a Gestlab para obtener los identificadores reales."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# id_peticion es una lista con los ID reales obtenidos tras la decodificación hexadecimal\n",
    "id_peticion = []\n",
    "for pdf in l_pdf:\n",
    "    # split por guion bajo\n",
    "    s_pdf = pdf.split('_')\n",
    "    # transformación del segundo componente en hexadecimal del identificador\n",
    "    hex_id = int(s_pdf[1],16)\n",
    "    id_peticion.append(hex_id)\n",
    "\n",
    "# ejecución de la función getIDs\n",
    "r_id = nimgenetics.getIDs(id_list=id_peticion)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 3.2.3 Almacenamiento de los informes con su ID correcto\n",
    "Selección de informes que cumplen un patrón específico: identifiedPDF.\n",
    "\n",
    "Informes cuyo identificador es desconocido: emptyPDF."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Generar una lista que almacene todos los identificadores provenientes de r_id\n",
    "# l_names a diferencia de la anterior no es una lista de listas\n",
    "l_names = []   \n",
    "for row in r_id:\n",
    "    try:\n",
    "        # Primer bloque: extraer identificadores que cumplan la siguiente estructura\n",
    "        l_names.append(row[0][0])\n",
    "    except:\n",
    "        # Segundo bloque: identificadores que devuelven elementos vacios\n",
    "        # Estos informes se almacenarán con su identificador hexadecimal original\n",
    "        s_row = str(row)\n",
    "        if s_row == '[]':\n",
    "            # Almacenamos en l_names los identificadores reemplazados con el literal 'sin_id'\n",
    "            sr_row = s_row.replace('[]', 'sin_id')\n",
    "            l_names.append(sr_row)\n",
    "\n",
    "\n",
    "# Función utilizada para almacenar los informes que cumplen un patrón de selección pre-establecido\n",
    "# También se almacenan los informes que están repetidos y los descartados que no cumplen el patrón\n",
    "# con la finalidad de no perder información\n",
    "nimgenetics.identifiedPDF(folder_path=folder_result, list_id=l_names, list_path=l_path)\n",
    "\n",
    "# Función para almacenar los informes cuyo identificador no ha sido recuperado de la base de datos\n",
    "# Estos informes son guardados con el identificador hexadecimal original\n",
    "nimgenetics.emptyPDF(folder_path=folder_result, list_id=l_names, list_path=l_path, list_pdf=l_pdf)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 3.3 Conversión de los informes a texto plano para la extracción de terminología médica"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# pdfPath contiene el path absoluto de todos los pdf \n",
    "# este paso es necesario para ejecutar el siguiente programa: pdfToImage\n",
    "pdfPath = nimgenetics.absolutePath(dpath=pathI)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 3.3.1 Generación de imágenes en formato PNG a partir de los informes escaneados\n",
    "Este procedimiento involucra la ejecución de la función pdfToImage."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Ejecución de la función pdfToImage\n",
    "nimgenetics.pdfToImage(f_path=folder_result, pdf_list=pdfPath)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 3.3.2 Generación de texto plano a partir de imágenes PNG\n",
    "Este procedimiento involucra la ejecución de dos funciones: makeCorpus y ocrToImage."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Ejecución de la función makeCorpus la cual integra en su código ocrToImage\n",
    "# Necesario image_path como parámetro para la correcta ejecución de la función\n",
    "# El resultado de este proceso guarda los resultados en la carpeta textos\n",
    "nimgenetics.makeCorpus(f_path=folder_result, image_path=image_path)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 4. EXPERIMENTOS REALIZADOS Y RESULTADOS\n",
    "\n",
    "Directorios importantes:\n",
    "\n",
    "- cutext_results: carpeta que incluye los resultados de la ejecución de CUTEXT.\n",
    "- cutext_corpus: carpeta que alberga el corpus elaborado al procesar los resultados de CUTEXT.\n",
    "- megadict: path donde se localiza el megadiccionario.\n",
    "- lookup_corpus: path donde almacenar el corpus generado por la estrategia lookup.\n",
    "- corpus_global: path donde guardar el corpus global elaborado a partir de cutext y lookup.\n",
    "- lookup_corpus_uncleaned: path donde se almacena el corpus de lookup sin proceso de curación."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cutext_results = 'path_to_save_cutext_results'\n",
    "cutext_corpus = 'path_to_save_cutext_corpus'\n",
    "megadict = 'path_to_megadict'\n",
    "lookup_corpus = 'path_to_lookup_corpus'\n",
    "corpus_global = 'path_to_merge_corpus'\n",
    "lookup_corpus_uncleaned = 'path_to_lookup_corpusUncleaned'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# path_textos contiene el path absoluto de los textos\n",
    "path_textos = nimgenetics.absolutePath(dpath=textos)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 4.2 Extracción de terminología mediante CUTEXT\n",
    "Ejecutar la función getCutextResults que integra la funcionalidad de CUTEXT."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# path a la carpeta in de cutext\n",
    "cutext_in = 'path_to_cutext_in_folder'\n",
    "# path del output generado por cutext\n",
    "out = 'path_to_cutext_output_folder'\n",
    "# El output de cutext se almacenará en cutext_results\n",
    "# ejecución de la función que integra la funcionalidad de cutext\n",
    "nimgenetics.getCutextResults(cutext_in=cutext_in, textos_ori=path_textos, old=out, new=cutext_results)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 4.2.1 Elaboración de un corpus final a partir de los resultados CUTEXT\n",
    "La función cutextCorpus procesa los resultados generados en el apartado anterior."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# path absoluto de los resultados de cutext\n",
    "l_cutext = nimgenetics.absolutePath(dpath=cutext_results)\n",
    "\n",
    "# Ejecución del programa que genera el corpus de cutext\n",
    "# El resultado se almacena en cutext_corpus\n",
    "nimgenetics.cutextCorpus(lista_cutext=l_cutext, path_to_words=cutext_corpus)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 4.3 Extracción de terminología mediante estrategia lookup\n",
    "\n",
    "La función lookup genera un corpus elaborado utilizando esta estrategia."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Almacenamos el corpus de lookup en lookup_corpus\n",
    "nimgenetics.lookup(textos_ori=path_textos, megadict=megadict, path_to_words=lookup_corpus)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 4.5 Elaboración de un corpus global a partir de cutext y lookup\n",
    "La función mergeCorpus implica la construcción de un corpus que alberga la terminología extraída mediante cutext y la estrategia lookup."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Este proceso genera el corpus global y se almacena en corpus_global\n",
    "nimgenetics.mergeCorpus(p_cutext=cutext_corpus, p_lookup=lookup_corpus, path_to_words=corpus_global)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 4.5.1 Representación de clínica sobre una matriz de distribución terminológica\n",
    "Diseñar una función capaz de mapear los términos extraídos al paciente correspondiente"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Construir la matriz a partir de un dataFrame almacenado como un documento excel\n",
    "nimgenetics.mapeoTerminos(l_textos=path_textos, infile_global=corpus_global)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 4.5.2 Selección de 500 términos para el mapeo manual a HPO\n",
    "Función que genera dos diccionarios relativo al conteo global de términos y los 500 términos para el mapeo manual a HPO."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_terms = nimgenetics.subsetManual(c_uncleaned=lookup_corpus_uncleaned, c_global=corpus_global)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "El programa devuelve el conteo global -*all_terms*- gracias a la función items del módulo *FreqDist* de NLTK. \n",
    "\n",
    "Esta lista es utilizada para generar la imagen de la distribución terminológia en función de la frecuencia de los términos."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# -all_terms- diccionario que integra la lista completa de términos\n",
    "# Los términos aparecen en formato key, value\n",
    "# key = término\n",
    "# value = número de veces que aparece el término\n",
    "\n",
    "# Primera categoría:\n",
    "values_500 = []\n",
    "values_500_50 = []\n",
    "values_50 = []\n",
    "\n",
    "for key, value in all_terms:\n",
    "    \n",
    "    if value > 500:\n",
    "        values_500.append(key)\n",
    "    elif value <= 500 and value >= 50:\n",
    "        values_500_50.append(key)\n",
    "    else:\n",
    "        values_50.append(key)\n",
    "\n",
    "names = ['500', '500-50', '50']\n",
    "values = [len(values_500), len(values_500_50), len(values_50)]\n",
    "\n",
    "# Segunda categoría:\n",
    "# Diccionario con las palabras <= 50 repeticiones en el corpus global\n",
    "dict2 = {k: v for k, v in all_terms if v <= 50}\n",
    "d2 = dict2.items()\n",
    "\n",
    "v_50_25 = []\n",
    "v_25_10 = []\n",
    "v_10 = []\n",
    "\n",
    "for key, value in d2:\n",
    "    \n",
    "    if value <= 50 and value > 25:\n",
    "        v_50_25.append(key)\n",
    "    elif value <= 25 and value > 10:\n",
    "        v25_10.append(key)\n",
    "    else:\n",
    "        v_10.append(key)\n",
    "\n",
    "names1 = ['50-25', '25-10', '10']\n",
    "values1 = [len(v_50_25), len(v_25_10), len(v_10)]\n",
    "\n",
    "# Tercera categoría:\n",
    "# Diccionario con las palabras <= 10 repeticiones en el corpus global\n",
    "dict3 = {k: v for k, v in all_terms if v <= 10}\n",
    "d3 = dict3.items()\n",
    "\n",
    "v_10_5 = []\n",
    "v_5_3 = []\n",
    "v_3 = []\n",
    "\n",
    "for key, value in d3:\n",
    "    \n",
    "    if value <= 10 and value > 5:\n",
    "        v_10_5.append(key)\n",
    "    elif value <= 5 and value > 3:\n",
    "        v_5_3.append(key)\n",
    "    else:\n",
    "        v_3.append(key)\n",
    "        \n",
    "names2 = ['10-5', '5-3', '3']\n",
    "values2 = [len(v_10_5), len(v_5_3), len(v_3)]\n",
    "\n",
    "# Cuarta categoría:\n",
    "# Diccionario con las palabras <= 3 repeticiones en el corpus global\n",
    "dict4 = {k: v for k, v in all_terms if v <= 3}\n",
    "d4 = dict4.items()\n",
    "\n",
    "v_3_ = []\n",
    "v_2_ = []\n",
    "v_1_ = []\n",
    "\n",
    "for key, value in d4:\n",
    "    \n",
    "    if value == 3:\n",
    "        v_3_.append(key)\n",
    "    elif value == 2:\n",
    "        v_2_.append(key)\n",
    "    else:\n",
    "        v_1_.append(key)\n",
    "\n",
    "names3 = ['3', '2', '1']\n",
    "values3 = [len(v_3_), len(v_2_), len(v_1_)]\n",
    "\n",
    "# Plotear la distribución de categorías terminológicas\n",
    "fig = plt.figure(figsize=(12,10))\n",
    "ax0 = fig.add_subplot(111)\n",
    "ax1 = fig.add_subplot(221)\n",
    "ax2 = fig.add_subplot(222)\n",
    "ax3 = fig.add_subplot(223)\n",
    "ax4 = fig.add_subplot(224)\n",
    "\n",
    "ax0.spines['top'].set_color('none')\n",
    "ax0.spines['bottom'].set_color('none')\n",
    "ax0.spines['left'].set_color('none')\n",
    "ax0.spines['right'].set_color('none')\n",
    "ax0.tick_params(labelcolor='w', top=False, bottom=False, left=False, right=False)\n",
    "\n",
    "ax0.set_ylabel('Conteo de términos', fontsize=12, labelpad=20)\n",
    "ax0.set_xlabel('Frecuencia de términos', fontsize=12, labelpad=20)\n",
    "\n",
    "ax1.bar(names, values)\n",
    "ax1.set_ylim(0, 4500)\n",
    "\n",
    "ax2.bar(names1, values1)\n",
    "ax2.set_ylim(0, 4500)\n",
    "\n",
    "ax3.bar(names2, values2)\n",
    "ax3.set_ylim(0, 4500)\n",
    "\n",
    "ax4.bar(names3, values3)\n",
    "ax4.set_ylim(0, 4500)\n",
    "\n",
    "fig.suptitle('Categorias Terminológicas', fontsize=20)\n",
    "fig.savefig('distribucion de terminología.png')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 4.6 EXOMISER\n",
    "\n",
    "Función que genera el template yml para poder ejecutar exomiser."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# path para el archivo yml y para guardar los resultados de la ejecución\n",
    "infile_template = 'path_to_yml_template'\n",
    "exo_path = 'path_to_save_exomiser_results'\n",
    "\n",
    "# Hacer el diccionario con los datos\n",
    "d = {'16NS-1': ['HP:0002715','HP:0002086'],\n",
    "     '16NS-2': ['HP:0002145','HP:0001251','HP:0002072','HP:0001260','HP:0002015','HP:0002127','HP:0002071',\n",
    "                'HP:0001272','HP:0002145','HP:0002527','HP:0002463','HP:0000666'],\n",
    "     '16NS-3': ['HP:0001644'],\n",
    "     '17NR-1': ['HP:0000729','HP:0001999','HP:0000286','HP:0000414','HP:0001611','HP:0000712'],\n",
    "     '17NR-2': ['HP:0002352','HP:0003429','HP:0002415','HP:0001252','HP:0012736','HP:0000577','HP:0030455',\n",
    "                'HP:0000338','HP:0000817','HP:0001762','HP:0002808','HP:0003376'],\n",
    "     '17NR-3': ['HP:0001263','HP:0007018','HP:0006919'],\n",
    "     '17NR-4': ['HP:0000118','HP:0000252','HP:0000532','HP:0001004','HP:0001363','HP:0010537','HP:0000233',\n",
    "                'HP:0000343','HP:0000581','HP:0025349','HP:0000488','HP:0001622'],\n",
    "     '18NR-1': ['HP:0000158','HP:0011410','HP:0002804','HP:0001298','HP:0001250','HP:0001274','HP:0001166',\n",
    "                'HP:0000119','HP:0001197'],\n",
    "     '18NR-2': ['HP:0004322','HP:0100021','HP:0001332','HP:0001263','HP:0001510','HP:0001250'],\n",
    "     '18NR-3': ['HP:0000924','HP:0000152','HP:0001507','HP:0040064','HP:0000707','HP:0000598','HP:0001574',\n",
    "                'HP:0001626','HP:0002715','HP:0002086','HP:0003011','HP:0001263','HP:0000403']}\n",
    "\n",
    "keys = list(d.keys())\n",
    "values = list(d.values())\n",
    "\n",
    "# ejecutar la función para preparar los templates para ejecutar exomiser\n",
    "nimgenetics.makeTemplate(patient_id=keys, lista_hpo=values, infile=infile_template, exomiser_path=exo_path)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 4.6.2 Resultados de la ejecución de exomiser y comparativa con Health29\n",
    "\n",
    "Lista de variantes candidatas obtenida a partir de la ejecución de exomiser."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 4.6.2.2 Caso 16NS-2\n",
    "\n",
    "Análisis del output del caso 16NS-2."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>functional_class</th>\n",
       "      <th>gene</th>\n",
       "      <th>variant_score</th>\n",
       "      <th>pheno_score</th>\n",
       "      <th>gene_variant_score</th>\n",
       "      <th>gene_combined_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chrX</td>\n",
       "      <td>56591569</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>UBQLN2</td>\n",
       "      <td>0.996715</td>\n",
       "      <td>0.787443</td>\n",
       "      <td>0.996715</td>\n",
       "      <td>0.982886</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr14</td>\n",
       "      <td>92537354</td>\n",
       "      <td>frameshift_elongation</td>\n",
       "      <td>ATXN3</td>\n",
       "      <td>0.950000</td>\n",
       "      <td>0.706157</td>\n",
       "      <td>0.950000</td>\n",
       "      <td>0.941401</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr16</td>\n",
       "      <td>730628</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>STUB1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.655719</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.937705</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr16</td>\n",
       "      <td>732171</td>\n",
       "      <td>frameshift_truncation</td>\n",
       "      <td>STUB1</td>\n",
       "      <td>0.950000</td>\n",
       "      <td>0.655719</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.937705</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr7</td>\n",
       "      <td>73795169</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>CLIP2</td>\n",
       "      <td>0.987958</td>\n",
       "      <td>0.657653</td>\n",
       "      <td>0.987958</td>\n",
       "      <td>0.932205</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr9</td>\n",
       "      <td>135218103</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>SETX</td>\n",
       "      <td>0.881138</td>\n",
       "      <td>0.749208</td>\n",
       "      <td>0.881138</td>\n",
       "      <td>0.930327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr6</td>\n",
       "      <td>170871049</td>\n",
       "      <td>disruptive_inframe_insertion</td>\n",
       "      <td>TBP</td>\n",
       "      <td>0.850000</td>\n",
       "      <td>0.764932</td>\n",
       "      <td>0.850000</td>\n",
       "      <td>0.921954</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr22</td>\n",
       "      <td>29886606</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NEFH</td>\n",
       "      <td>0.997507</td>\n",
       "      <td>0.614849</td>\n",
       "      <td>0.997507</td>\n",
       "      <td>0.905837</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr9</td>\n",
       "      <td>140773612</td>\n",
       "      <td>splice_donor_variant</td>\n",
       "      <td>CACNA1B</td>\n",
       "      <td>0.900000</td>\n",
       "      <td>0.605219</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.899043</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr9</td>\n",
       "      <td>140777306</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>CACNA1B</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.605219</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.899043</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      chr        pos              functional_class     gene  variant_score  \\\n",
       "0    chrX   56591569              missense_variant   UBQLN2       0.996715   \n",
       "1   chr14   92537354         frameshift_elongation    ATXN3       0.950000   \n",
       "2   chr16     730628              missense_variant    STUB1       1.000000   \n",
       "4   chr16     732171         frameshift_truncation    STUB1       0.950000   \n",
       "5    chr7   73795169              missense_variant    CLIP2       0.987958   \n",
       "6    chr9  135218103              missense_variant     SETX       0.881138   \n",
       "7    chr6  170871049  disruptive_inframe_insertion      TBP       0.850000   \n",
       "9   chr22   29886606              missense_variant     NEFH       0.997507   \n",
       "10   chr9  140773612          splice_donor_variant  CACNA1B       0.900000   \n",
       "11   chr9  140777306              missense_variant  CACNA1B       1.000000   \n",
       "\n",
       "    pheno_score  gene_variant_score  gene_combined_score  \n",
       "0      0.787443            0.996715             0.982886  \n",
       "1      0.706157            0.950000             0.941401  \n",
       "2      0.655719            1.000000             0.937705  \n",
       "4      0.655719            1.000000             0.937705  \n",
       "5      0.657653            0.987958             0.932205  \n",
       "6      0.749208            0.881138             0.930327  \n",
       "7      0.764932            0.850000             0.921954  \n",
       "9      0.614849            0.997507             0.905837  \n",
       "10     0.605219            1.000000             0.899043  \n",
       "11     0.605219            1.000000             0.899043  "
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "caso2 = 'path_to_16NS-2_output.variants.tsv'\n",
    "nimgenetics.filterVariants(data=caso2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Posición de la variante genética indicada por los analistas en la lista de genes generada por exomiser."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>functional_class</th>\n",
       "      <th>gene</th>\n",
       "      <th>variant_score</th>\n",
       "      <th>pheno_score</th>\n",
       "      <th>gene_variant_score</th>\n",
       "      <th>gene_combined_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chrX</td>\n",
       "      <td>56591569</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>UBQLN2</td>\n",
       "      <td>0.996715</td>\n",
       "      <td>0.787443</td>\n",
       "      <td>0.996715</td>\n",
       "      <td>0.982886</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chr       pos  functional_class    gene  variant_score  pheno_score  \\\n",
       "0  chrX  56591569  missense_variant  UBQLN2       0.996715     0.787443   \n",
       "\n",
       "   gene_variant_score  gene_combined_score  \n",
       "0            0.996715             0.982886  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nimgenetics.locVariant(data=caso2, gen_sel='UBQLN2')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 4.6.2.3 Caso 16NS-3\n",
    "\n",
    "Análisis del output del caso 16NS-3."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>functional_class</th>\n",
       "      <th>gene</th>\n",
       "      <th>variant_score</th>\n",
       "      <th>pheno_score</th>\n",
       "      <th>gene_variant_score</th>\n",
       "      <th>gene_combined_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23899016</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MYH7</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.998149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>69881820</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MYPN</td>\n",
       "      <td>0.999596</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.999596</td>\n",
       "      <td>0.998142</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr2</td>\n",
       "      <td>179600304</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>TTN</td>\n",
       "      <td>0.992592</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.992592</td>\n",
       "      <td>0.998019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr2</td>\n",
       "      <td>179640548</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>TTN</td>\n",
       "      <td>0.987838</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.992592</td>\n",
       "      <td>0.998019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chrX</td>\n",
       "      <td>31747837</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>DMD</td>\n",
       "      <td>0.922496</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.922496</td>\n",
       "      <td>0.996236</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr5</td>\n",
       "      <td>137222922</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MYOT</td>\n",
       "      <td>0.997149</td>\n",
       "      <td>0.829720</td>\n",
       "      <td>0.997149</td>\n",
       "      <td>0.988948</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr4</td>\n",
       "      <td>996513</td>\n",
       "      <td>splice_region_variant</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>0.900000</td>\n",
       "      <td>0.766280</td>\n",
       "      <td>0.900000</td>\n",
       "      <td>0.949903</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr6</td>\n",
       "      <td>110064448</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>FIG4</td>\n",
       "      <td>0.999797</td>\n",
       "      <td>0.671483</td>\n",
       "      <td>0.999797</td>\n",
       "      <td>0.946524</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr6</td>\n",
       "      <td>31238930</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>HLA-B</td>\n",
       "      <td>0.758000</td>\n",
       "      <td>0.641241</td>\n",
       "      <td>0.997000</td>\n",
       "      <td>0.926458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr6</td>\n",
       "      <td>31239006</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>HLA-B</td>\n",
       "      <td>0.997000</td>\n",
       "      <td>0.641241</td>\n",
       "      <td>0.997000</td>\n",
       "      <td>0.926458</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      chr        pos       functional_class   gene  variant_score  \\\n",
       "1   chr14   23899016       missense_variant   MYH7       1.000000   \n",
       "2   chr10   69881820       missense_variant   MYPN       0.999596   \n",
       "4    chr2  179600304       missense_variant    TTN       0.992592   \n",
       "6    chr2  179640548       missense_variant    TTN       0.987838   \n",
       "7    chrX   31747837       missense_variant    DMD       0.922496   \n",
       "9    chr5  137222922       missense_variant   MYOT       0.997149   \n",
       "10   chr4     996513  splice_region_variant   IDUA       0.900000   \n",
       "11   chr6  110064448       missense_variant   FIG4       0.999797   \n",
       "15   chr6   31238930       missense_variant  HLA-B       0.758000   \n",
       "16   chr6   31239006       missense_variant  HLA-B       0.997000   \n",
       "\n",
       "    pheno_score  gene_variant_score  gene_combined_score  \n",
       "1      1.000000            1.000000             0.998149  \n",
       "2      1.000000            0.999596             0.998142  \n",
       "4      1.000000            0.992592             0.998019  \n",
       "6      1.000000            0.992592             0.998019  \n",
       "7      1.000000            0.922496             0.996236  \n",
       "9      0.829720            0.997149             0.988948  \n",
       "10     0.766280            0.900000             0.949903  \n",
       "11     0.671483            0.999797             0.946524  \n",
       "15     0.641241            0.997000             0.926458  \n",
       "16     0.641241            0.997000             0.926458  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "caso3 = 'path_to_16NS-3_output.variants.tsv'\n",
    "nimgenetics.filterVariants(data=caso3)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Posición de la variante genética indicada por los analistas en la lista de genes generada por exomiser."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>functional_class</th>\n",
       "      <th>gene</th>\n",
       "      <th>variant_score</th>\n",
       "      <th>pheno_score</th>\n",
       "      <th>gene_variant_score</th>\n",
       "      <th>gene_combined_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23892645</td>\n",
       "      <td>coding_transcript_intron_variant</td>\n",
       "      <td>MYH7</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.998149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23899016</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MYH7</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.998149</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     chr       pos                  functional_class  gene  variant_score  \\\n",
       "0  chr14  23892645  coding_transcript_intron_variant  MYH7            0.0   \n",
       "1  chr14  23899016                  missense_variant  MYH7            1.0   \n",
       "\n",
       "   pheno_score  gene_variant_score  gene_combined_score  \n",
       "0          1.0                 1.0             0.998149  \n",
       "1          1.0                 1.0             0.998149  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nimgenetics.locVariant(data=caso3, gen_sel='MYH7')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 4.6.2.4 Caso 17NR-1\n",
    "\n",
    "Análisis del output del caso 17NR-1."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>functional_class</th>\n",
       "      <th>gene</th>\n",
       "      <th>variant_score</th>\n",
       "      <th>pheno_score</th>\n",
       "      <th>gene_variant_score</th>\n",
       "      <th>gene_combined_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>162</th>\n",
       "      <td>chr7</td>\n",
       "      <td>73477660</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>ELN</td>\n",
       "      <td>0.994485</td>\n",
       "      <td>0.735234</td>\n",
       "      <td>0.994485</td>\n",
       "      <td>0.970328</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>163</th>\n",
       "      <td>chr22</td>\n",
       "      <td>24109722</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>CHCHD10</td>\n",
       "      <td>0.912603</td>\n",
       "      <td>0.757119</td>\n",
       "      <td>0.912603</td>\n",
       "      <td>0.950871</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>164</th>\n",
       "      <td>chr11</td>\n",
       "      <td>65978677</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>PACS1</td>\n",
       "      <td>0.998427</td>\n",
       "      <td>0.680921</td>\n",
       "      <td>0.998427</td>\n",
       "      <td>0.950691</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>166</th>\n",
       "      <td>chr16</td>\n",
       "      <td>49672716</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>ZNF423</td>\n",
       "      <td>0.998423</td>\n",
       "      <td>0.651352</td>\n",
       "      <td>0.998423</td>\n",
       "      <td>0.934115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>167</th>\n",
       "      <td>chr18</td>\n",
       "      <td>48581231</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>SMAD4</td>\n",
       "      <td>0.997270</td>\n",
       "      <td>0.643719</td>\n",
       "      <td>0.997270</td>\n",
       "      <td>0.928359</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>168</th>\n",
       "      <td>chr22</td>\n",
       "      <td>51121852</td>\n",
       "      <td>splice_region_variant</td>\n",
       "      <td>SHANK3</td>\n",
       "      <td>0.811295</td>\n",
       "      <td>0.788879</td>\n",
       "      <td>0.811295</td>\n",
       "      <td>0.913936</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>169</th>\n",
       "      <td>chr12</td>\n",
       "      <td>112036767</td>\n",
       "      <td>disruptive_inframe_insertion</td>\n",
       "      <td>ATXN2</td>\n",
       "      <td>0.845490</td>\n",
       "      <td>0.757119</td>\n",
       "      <td>0.845490</td>\n",
       "      <td>0.912661</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170</th>\n",
       "      <td>chr14</td>\n",
       "      <td>73659482</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>PSEN1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.615655</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.908472</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>171</th>\n",
       "      <td>chr6</td>\n",
       "      <td>32026164</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>TNXB</td>\n",
       "      <td>0.992000</td>\n",
       "      <td>0.622534</td>\n",
       "      <td>0.992000</td>\n",
       "      <td>0.908308</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>173</th>\n",
       "      <td>chr1</td>\n",
       "      <td>103496754</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>COL11A1</td>\n",
       "      <td>0.991646</td>\n",
       "      <td>0.622644</td>\n",
       "      <td>0.991646</td>\n",
       "      <td>0.908132</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       chr        pos              functional_class     gene  variant_score  \\\n",
       "162   chr7   73477660              missense_variant      ELN       0.994485   \n",
       "163  chr22   24109722              missense_variant  CHCHD10       0.912603   \n",
       "164  chr11   65978677              missense_variant    PACS1       0.998427   \n",
       "166  chr16   49672716              missense_variant   ZNF423       0.998423   \n",
       "167  chr18   48581231              missense_variant    SMAD4       0.997270   \n",
       "168  chr22   51121852         splice_region_variant   SHANK3       0.811295   \n",
       "169  chr12  112036767  disruptive_inframe_insertion    ATXN2       0.845490   \n",
       "170  chr14   73659482              missense_variant    PSEN1       1.000000   \n",
       "171   chr6   32026164              missense_variant     TNXB       0.992000   \n",
       "173   chr1  103496754              missense_variant  COL11A1       0.991646   \n",
       "\n",
       "     pheno_score  gene_variant_score  gene_combined_score  \n",
       "162     0.735234            0.994485             0.970328  \n",
       "163     0.757119            0.912603             0.950871  \n",
       "164     0.680921            0.998427             0.950691  \n",
       "166     0.651352            0.998423             0.934115  \n",
       "167     0.643719            0.997270             0.928359  \n",
       "168     0.788879            0.811295             0.913936  \n",
       "169     0.757119            0.845490             0.912661  \n",
       "170     0.615655            1.000000             0.908472  \n",
       "171     0.622534            0.992000             0.908308  \n",
       "173     0.622644            0.991646             0.908132  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "caso4 = 'path_to_17NR-1_output.variants.tsv'\n",
    "nimgenetics.filterVariants(data=caso4)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Posición de la variante genética indicada por los analistas en la lista de genes generada por exomiser."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>functional_class</th>\n",
       "      <th>gene</th>\n",
       "      <th>variant_score</th>\n",
       "      <th>pheno_score</th>\n",
       "      <th>gene_variant_score</th>\n",
       "      <th>gene_combined_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>164</th>\n",
       "      <td>chr11</td>\n",
       "      <td>65978677</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>PACS1</td>\n",
       "      <td>0.998427</td>\n",
       "      <td>0.680921</td>\n",
       "      <td>0.998427</td>\n",
       "      <td>0.950691</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165</th>\n",
       "      <td>chr11</td>\n",
       "      <td>66010762</td>\n",
       "      <td>3_prime_UTR_exon_variant</td>\n",
       "      <td>PACS1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.680921</td>\n",
       "      <td>0.998427</td>\n",
       "      <td>0.950691</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       chr       pos          functional_class   gene  variant_score  \\\n",
       "164  chr11  65978677          missense_variant  PACS1       0.998427   \n",
       "165  chr11  66010762  3_prime_UTR_exon_variant  PACS1       0.000000   \n",
       "\n",
       "     pheno_score  gene_variant_score  gene_combined_score  \n",
       "164     0.680921            0.998427             0.950691  \n",
       "165     0.680921            0.998427             0.950691  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nimgenetics.locVariant(data=caso4, gen_sel='PACS1')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 4.6.2.5 Caso 17NR-2\n",
    "\n",
    "Análisis del output del caso 17NR-2."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>functional_class</th>\n",
       "      <th>gene</th>\n",
       "      <th>variant_score</th>\n",
       "      <th>pheno_score</th>\n",
       "      <th>gene_variant_score</th>\n",
       "      <th>gene_combined_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr3</td>\n",
       "      <td>132384868</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>UBA5</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.656829</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.938376</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr3</td>\n",
       "      <td>132394747</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>UBA5</td>\n",
       "      <td>0.888318</td>\n",
       "      <td>0.656829</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.938376</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr12</td>\n",
       "      <td>7361645</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>PEX5</td>\n",
       "      <td>0.983341</td>\n",
       "      <td>0.653448</td>\n",
       "      <td>0.983341</td>\n",
       "      <td>0.926557</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr5</td>\n",
       "      <td>60368955</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NDUFAF2</td>\n",
       "      <td>0.994657</td>\n",
       "      <td>0.634968</td>\n",
       "      <td>0.994657</td>\n",
       "      <td>0.920326</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr19</td>\n",
       "      <td>39911625</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>PLEKHG2</td>\n",
       "      <td>0.992780</td>\n",
       "      <td>0.631986</td>\n",
       "      <td>0.992780</td>\n",
       "      <td>0.916717</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr4</td>\n",
       "      <td>3076650</td>\n",
       "      <td>frameshift_truncation</td>\n",
       "      <td>HTT</td>\n",
       "      <td>0.950000</td>\n",
       "      <td>0.633110</td>\n",
       "      <td>0.950000</td>\n",
       "      <td>0.882608</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr4</td>\n",
       "      <td>3076653</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>HTT</td>\n",
       "      <td>0.950000</td>\n",
       "      <td>0.633110</td>\n",
       "      <td>0.950000</td>\n",
       "      <td>0.882608</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr4</td>\n",
       "      <td>3076659</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>HTT</td>\n",
       "      <td>0.785737</td>\n",
       "      <td>0.633110</td>\n",
       "      <td>0.950000</td>\n",
       "      <td>0.882608</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr4</td>\n",
       "      <td>3076662</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>HTT</td>\n",
       "      <td>0.759000</td>\n",
       "      <td>0.633110</td>\n",
       "      <td>0.950000</td>\n",
       "      <td>0.882608</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr4</td>\n",
       "      <td>3076665</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>HTT</td>\n",
       "      <td>0.644328</td>\n",
       "      <td>0.633110</td>\n",
       "      <td>0.950000</td>\n",
       "      <td>0.882608</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     chr        pos       functional_class     gene  variant_score  \\\n",
       "0   chr3  132384868       missense_variant     UBA5       1.000000   \n",
       "1   chr3  132394747       missense_variant     UBA5       0.888318   \n",
       "2  chr12    7361645       missense_variant     PEX5       0.983341   \n",
       "3   chr5   60368955       missense_variant  NDUFAF2       0.994657   \n",
       "4  chr19   39911625       missense_variant  PLEKHG2       0.992780   \n",
       "5   chr4    3076650  frameshift_truncation      HTT       0.950000   \n",
       "6   chr4    3076653     frameshift_variant      HTT       0.950000   \n",
       "7   chr4    3076659       missense_variant      HTT       0.785737   \n",
       "8   chr4    3076662       missense_variant      HTT       0.759000   \n",
       "9   chr4    3076665       missense_variant      HTT       0.644328   \n",
       "\n",
       "   pheno_score  gene_variant_score  gene_combined_score  \n",
       "0     0.656829            1.000000             0.938376  \n",
       "1     0.656829            1.000000             0.938376  \n",
       "2     0.653448            0.983341             0.926557  \n",
       "3     0.634968            0.994657             0.920326  \n",
       "4     0.631986            0.992780             0.916717  \n",
       "5     0.633110            0.950000             0.882608  \n",
       "6     0.633110            0.950000             0.882608  \n",
       "7     0.633110            0.950000             0.882608  \n",
       "8     0.633110            0.950000             0.882608  \n",
       "9     0.633110            0.950000             0.882608  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "caso5 = 'path_to_17NR-2_output.variants.tsv'\n",
    "nimgenetics.filterVariants(data=caso5)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Posición de la variante genética indicada por los analistas en la lista de genes generada por exomiser."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>functional_class</th>\n",
       "      <th>gene</th>\n",
       "      <th>variant_score</th>\n",
       "      <th>pheno_score</th>\n",
       "      <th>gene_variant_score</th>\n",
       "      <th>gene_combined_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr3</td>\n",
       "      <td>132384868</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>UBA5</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.656829</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.938376</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr3</td>\n",
       "      <td>132394747</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>UBA5</td>\n",
       "      <td>0.888318</td>\n",
       "      <td>0.656829</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.938376</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chr        pos  functional_class  gene  variant_score  pheno_score  \\\n",
       "0  chr3  132384868  missense_variant  UBA5       1.000000     0.656829   \n",
       "1  chr3  132394747  missense_variant  UBA5       0.888318     0.656829   \n",
       "\n",
       "   gene_variant_score  gene_combined_score  \n",
       "0                 1.0             0.938376  \n",
       "1                 1.0             0.938376  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nimgenetics.locVariant(data=caso5, gen_sel='UBA5')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 4.6.2.6 Caso 17NR-3\n",
    "\n",
    "Análisis del output del caso 17NR-3."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>functional_class</th>\n",
       "      <th>gene</th>\n",
       "      <th>variant_score</th>\n",
       "      <th>pheno_score</th>\n",
       "      <th>gene_variant_score</th>\n",
       "      <th>gene_combined_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>chr3</td>\n",
       "      <td>9517375</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>SETD5</td>\n",
       "      <td>0.931024</td>\n",
       "      <td>0.904502</td>\n",
       "      <td>0.931024</td>\n",
       "      <td>0.990660</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>224</th>\n",
       "      <td>chr7</td>\n",
       "      <td>73482987</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>ELN</td>\n",
       "      <td>0.902591</td>\n",
       "      <td>0.895430</td>\n",
       "      <td>0.902591</td>\n",
       "      <td>0.986727</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>225</th>\n",
       "      <td>chr6</td>\n",
       "      <td>170871051</td>\n",
       "      <td>frameshift_truncation</td>\n",
       "      <td>TBP</td>\n",
       "      <td>0.950000</td>\n",
       "      <td>0.819662</td>\n",
       "      <td>0.950000</td>\n",
       "      <td>0.981229</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>226</th>\n",
       "      <td>chr6</td>\n",
       "      <td>170871054</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>TBP</td>\n",
       "      <td>0.950000</td>\n",
       "      <td>0.819662</td>\n",
       "      <td>0.950000</td>\n",
       "      <td>0.981229</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>228</th>\n",
       "      <td>chr1</td>\n",
       "      <td>151400625</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>POGZ</td>\n",
       "      <td>0.950000</td>\n",
       "      <td>0.817993</td>\n",
       "      <td>0.950000</td>\n",
       "      <td>0.980906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>229</th>\n",
       "      <td>chr16</td>\n",
       "      <td>2112607</td>\n",
       "      <td>splice_region_variant</td>\n",
       "      <td>TSC2</td>\n",
       "      <td>0.900000</td>\n",
       "      <td>0.787658</td>\n",
       "      <td>0.900000</td>\n",
       "      <td>0.959481</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>230</th>\n",
       "      <td>chr9</td>\n",
       "      <td>140773612</td>\n",
       "      <td>splice_donor_variant</td>\n",
       "      <td>CACNA1B</td>\n",
       "      <td>0.900000</td>\n",
       "      <td>0.650247</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.934299</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>231</th>\n",
       "      <td>chr9</td>\n",
       "      <td>140777306</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>CACNA1B</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.650247</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.934299</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233</th>\n",
       "      <td>chr4</td>\n",
       "      <td>23816235</td>\n",
       "      <td>splice_region_variant</td>\n",
       "      <td>PPARGC1A</td>\n",
       "      <td>0.900000</td>\n",
       "      <td>0.728623</td>\n",
       "      <td>0.900000</td>\n",
       "      <td>0.927639</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>234</th>\n",
       "      <td>chr1</td>\n",
       "      <td>110256115</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>GSTM5</td>\n",
       "      <td>0.994000</td>\n",
       "      <td>0.641622</td>\n",
       "      <td>0.994000</td>\n",
       "      <td>0.924835</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       chr        pos       functional_class      gene  variant_score  \\\n",
       "34    chr3    9517375       missense_variant     SETD5       0.931024   \n",
       "224   chr7   73482987       missense_variant       ELN       0.902591   \n",
       "225   chr6  170871051  frameshift_truncation       TBP       0.950000   \n",
       "226   chr6  170871054     frameshift_variant       TBP       0.950000   \n",
       "228   chr1  151400625            stop_gained      POGZ       0.950000   \n",
       "229  chr16    2112607  splice_region_variant      TSC2       0.900000   \n",
       "230   chr9  140773612   splice_donor_variant   CACNA1B       0.900000   \n",
       "231   chr9  140777306       missense_variant   CACNA1B       1.000000   \n",
       "233   chr4   23816235  splice_region_variant  PPARGC1A       0.900000   \n",
       "234   chr1  110256115       missense_variant     GSTM5       0.994000   \n",
       "\n",
       "     pheno_score  gene_variant_score  gene_combined_score  \n",
       "34      0.904502            0.931024             0.990660  \n",
       "224     0.895430            0.902591             0.986727  \n",
       "225     0.819662            0.950000             0.981229  \n",
       "226     0.819662            0.950000             0.981229  \n",
       "228     0.817993            0.950000             0.980906  \n",
       "229     0.787658            0.900000             0.959481  \n",
       "230     0.650247            1.000000             0.934299  \n",
       "231     0.650247            1.000000             0.934299  \n",
       "233     0.728623            0.900000             0.927639  \n",
       "234     0.641622            0.994000             0.924835  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "caso6 = 'path_to_17NR-3_output.variants.tsv'\n",
    "nimgenetics.filterVariants(data=caso6)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Posición de la variante genética indicada por los analistas en la lista de genes generada por exomiser."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>functional_class</th>\n",
       "      <th>gene</th>\n",
       "      <th>variant_score</th>\n",
       "      <th>pheno_score</th>\n",
       "      <th>gene_variant_score</th>\n",
       "      <th>gene_combined_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>227</th>\n",
       "      <td>chr1</td>\n",
       "      <td>151384338</td>\n",
       "      <td>coding_transcript_intron_variant</td>\n",
       "      <td>POGZ</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.817993</td>\n",
       "      <td>0.95</td>\n",
       "      <td>0.980906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>228</th>\n",
       "      <td>chr1</td>\n",
       "      <td>151400625</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>POGZ</td>\n",
       "      <td>0.95</td>\n",
       "      <td>0.817993</td>\n",
       "      <td>0.95</td>\n",
       "      <td>0.980906</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      chr        pos                  functional_class  gene  variant_score  \\\n",
       "227  chr1  151384338  coding_transcript_intron_variant  POGZ           0.00   \n",
       "228  chr1  151400625                       stop_gained  POGZ           0.95   \n",
       "\n",
       "     pheno_score  gene_variant_score  gene_combined_score  \n",
       "227     0.817993                0.95             0.980906  \n",
       "228     0.817993                0.95             0.980906  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nimgenetics.locVariant(data=caso6, gen_sel='POGZ')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 4.6.2.7 Caso 17NR-4\n",
    "\n",
    "Análisis del output del caso 17NR-4."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>functional_class</th>\n",
       "      <th>gene</th>\n",
       "      <th>variant_score</th>\n",
       "      <th>pheno_score</th>\n",
       "      <th>gene_variant_score</th>\n",
       "      <th>gene_combined_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>97402858</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>ALDH18A1</td>\n",
       "      <td>0.984244</td>\n",
       "      <td>0.645947</td>\n",
       "      <td>0.984244</td>\n",
       "      <td>0.921672</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>169</th>\n",
       "      <td>chr10</td>\n",
       "      <td>94410275</td>\n",
       "      <td>splice_donor_variant</td>\n",
       "      <td>KIF11</td>\n",
       "      <td>0.900000</td>\n",
       "      <td>0.704208</td>\n",
       "      <td>0.900000</td>\n",
       "      <td>0.908645</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>174</th>\n",
       "      <td>chr6</td>\n",
       "      <td>31239006</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>HLA-B</td>\n",
       "      <td>0.997000</td>\n",
       "      <td>0.577401</td>\n",
       "      <td>0.997000</td>\n",
       "      <td>0.866450</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>177</th>\n",
       "      <td>chr6</td>\n",
       "      <td>31323202</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>HLA-B</td>\n",
       "      <td>0.001000</td>\n",
       "      <td>0.577401</td>\n",
       "      <td>0.997000</td>\n",
       "      <td>0.866450</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>178</th>\n",
       "      <td>chr6</td>\n",
       "      <td>31323962</td>\n",
       "      <td>frameshift_elongation</td>\n",
       "      <td>HLA-B</td>\n",
       "      <td>0.950000</td>\n",
       "      <td>0.577401</td>\n",
       "      <td>0.997000</td>\n",
       "      <td>0.866450</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>179</th>\n",
       "      <td>chr6</td>\n",
       "      <td>31324100</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>HLA-B</td>\n",
       "      <td>0.488000</td>\n",
       "      <td>0.577401</td>\n",
       "      <td>0.997000</td>\n",
       "      <td>0.866450</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>181</th>\n",
       "      <td>chr1</td>\n",
       "      <td>120510194</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NOTCH2</td>\n",
       "      <td>0.835154</td>\n",
       "      <td>0.625004</td>\n",
       "      <td>0.942000</td>\n",
       "      <td>0.865254</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>183</th>\n",
       "      <td>chr1</td>\n",
       "      <td>120572547</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NOTCH2</td>\n",
       "      <td>0.871000</td>\n",
       "      <td>0.625004</td>\n",
       "      <td>0.942000</td>\n",
       "      <td>0.865254</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>184</th>\n",
       "      <td>chr1</td>\n",
       "      <td>120572572</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NOTCH2</td>\n",
       "      <td>0.942000</td>\n",
       "      <td>0.625004</td>\n",
       "      <td>0.942000</td>\n",
       "      <td>0.865254</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193</th>\n",
       "      <td>chr1</td>\n",
       "      <td>120611964</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NOTCH2</td>\n",
       "      <td>0.815000</td>\n",
       "      <td>0.625004</td>\n",
       "      <td>0.942000</td>\n",
       "      <td>0.865254</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       chr        pos       functional_class      gene  variant_score  \\\n",
       "0    chr10   97402858       missense_variant  ALDH18A1       0.984244   \n",
       "169  chr10   94410275   splice_donor_variant     KIF11       0.900000   \n",
       "174   chr6   31239006       missense_variant     HLA-B       0.997000   \n",
       "177   chr6   31323202       missense_variant     HLA-B       0.001000   \n",
       "178   chr6   31323962  frameshift_elongation     HLA-B       0.950000   \n",
       "179   chr6   31324100       missense_variant     HLA-B       0.488000   \n",
       "181   chr1  120510194       missense_variant    NOTCH2       0.835154   \n",
       "183   chr1  120572547       missense_variant    NOTCH2       0.871000   \n",
       "184   chr1  120572572       missense_variant    NOTCH2       0.942000   \n",
       "193   chr1  120611964       missense_variant    NOTCH2       0.815000   \n",
       "\n",
       "     pheno_score  gene_variant_score  gene_combined_score  \n",
       "0       0.645947            0.984244             0.921672  \n",
       "169     0.704208            0.900000             0.908645  \n",
       "174     0.577401            0.997000             0.866450  \n",
       "177     0.577401            0.997000             0.866450  \n",
       "178     0.577401            0.997000             0.866450  \n",
       "179     0.577401            0.997000             0.866450  \n",
       "181     0.625004            0.942000             0.865254  \n",
       "183     0.625004            0.942000             0.865254  \n",
       "184     0.625004            0.942000             0.865254  \n",
       "193     0.625004            0.942000             0.865254  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "caso7 = 'path_to_17NR-4_output.variants.tsv'\n",
    "nimgenetics.filterVariants(data=caso7)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Posición de la variante genética indicada por los analistas en la lista de genes generada por exomiser."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>functional_class</th>\n",
       "      <th>gene</th>\n",
       "      <th>variant_score</th>\n",
       "      <th>pheno_score</th>\n",
       "      <th>gene_variant_score</th>\n",
       "      <th>gene_combined_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>169</th>\n",
       "      <td>chr10</td>\n",
       "      <td>94410275</td>\n",
       "      <td>splice_donor_variant</td>\n",
       "      <td>KIF11</td>\n",
       "      <td>0.9</td>\n",
       "      <td>0.704208</td>\n",
       "      <td>0.9</td>\n",
       "      <td>0.908645</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       chr       pos      functional_class   gene  variant_score  pheno_score  \\\n",
       "169  chr10  94410275  splice_donor_variant  KIF11            0.9     0.704208   \n",
       "\n",
       "     gene_variant_score  gene_combined_score  \n",
       "169                 0.9             0.908645  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nimgenetics.locVariant(data=caso7, gen_sel='KIF11')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Caso 18NR-1\n",
    "\n",
    "Análisis del output del caso 18NR-1."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>functional_class</th>\n",
       "      <th>gene</th>\n",
       "      <th>variant_score</th>\n",
       "      <th>pheno_score</th>\n",
       "      <th>gene_variant_score</th>\n",
       "      <th>gene_combined_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr18</td>\n",
       "      <td>42531907</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>SETBP1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.673894</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.947870</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>8420514</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>RERE</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.622745</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.914418</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>6111709</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>KCNAB2</td>\n",
       "      <td>0.981009</td>\n",
       "      <td>0.622745</td>\n",
       "      <td>0.981009</td>\n",
       "      <td>0.899745</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chrX</td>\n",
       "      <td>132888207</td>\n",
       "      <td>splice_region_variant</td>\n",
       "      <td>GPC3</td>\n",
       "      <td>0.900000</td>\n",
       "      <td>0.693116</td>\n",
       "      <td>0.900000</td>\n",
       "      <td>0.898613</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr2</td>\n",
       "      <td>152420160</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NEB</td>\n",
       "      <td>0.717769</td>\n",
       "      <td>0.626933</td>\n",
       "      <td>0.959208</td>\n",
       "      <td>0.884700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr2</td>\n",
       "      <td>152584282</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NEB</td>\n",
       "      <td>0.959208</td>\n",
       "      <td>0.626933</td>\n",
       "      <td>0.959208</td>\n",
       "      <td>0.884700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr6</td>\n",
       "      <td>3225770</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>TUBB2B</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.589064</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.882748</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr22</td>\n",
       "      <td>21153431</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>PI4KA</td>\n",
       "      <td>0.982848</td>\n",
       "      <td>0.593527</td>\n",
       "      <td>0.982848</td>\n",
       "      <td>0.870747</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr10</td>\n",
       "      <td>101487264</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>COX15</td>\n",
       "      <td>0.980254</td>\n",
       "      <td>0.595636</td>\n",
       "      <td>0.980254</td>\n",
       "      <td>0.870533</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr6</td>\n",
       "      <td>142691950</td>\n",
       "      <td>frameshift_truncation</td>\n",
       "      <td>ADGRG6</td>\n",
       "      <td>0.950000</td>\n",
       "      <td>0.619586</td>\n",
       "      <td>0.950000</td>\n",
       "      <td>0.867242</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      chr        pos       functional_class    gene  variant_score  \\\n",
       "0   chr18   42531907       missense_variant  SETBP1       1.000000   \n",
       "1    chr1    8420514       missense_variant    RERE       1.000000   \n",
       "5    chr1    6111709       missense_variant  KCNAB2       0.981009   \n",
       "7    chrX  132888207  splice_region_variant    GPC3       0.900000   \n",
       "9    chr2  152420160       missense_variant     NEB       0.717769   \n",
       "11   chr2  152584282       missense_variant     NEB       0.959208   \n",
       "12   chr6    3225770       missense_variant  TUBB2B       1.000000   \n",
       "13  chr22   21153431       missense_variant   PI4KA       0.982848   \n",
       "14  chr10  101487264       missense_variant   COX15       0.980254   \n",
       "16   chr6  142691950  frameshift_truncation  ADGRG6       0.950000   \n",
       "\n",
       "    pheno_score  gene_variant_score  gene_combined_score  \n",
       "0      0.673894            1.000000             0.947870  \n",
       "1      0.622745            1.000000             0.914418  \n",
       "5      0.622745            0.981009             0.899745  \n",
       "7      0.693116            0.900000             0.898613  \n",
       "9      0.626933            0.959208             0.884700  \n",
       "11     0.626933            0.959208             0.884700  \n",
       "12     0.589064            1.000000             0.882748  \n",
       "13     0.593527            0.982848             0.870747  \n",
       "14     0.595636            0.980254             0.870533  \n",
       "16     0.619586            0.950000             0.867242  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "caso8 = 'path_to_18NR-1_output.variants.tsv'\n",
    "nimgenetics.filterVariants(data=caso8)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Posición de la variante genética indicada por los analistas en la lista de genes generada por exomiser."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>functional_class</th>\n",
       "      <th>gene</th>\n",
       "      <th>variant_score</th>\n",
       "      <th>pheno_score</th>\n",
       "      <th>gene_variant_score</th>\n",
       "      <th>gene_combined_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr18</td>\n",
       "      <td>42531907</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>SETBP1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.673894</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.94787</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     chr       pos  functional_class    gene  variant_score  pheno_score  \\\n",
       "0  chr18  42531907  missense_variant  SETBP1            1.0     0.673894   \n",
       "\n",
       "   gene_variant_score  gene_combined_score  \n",
       "0                 1.0              0.94787  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nimgenetics.locVariant(data=caso8, gen_sel='SETBP1')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Caso 18NR-2\n",
    "\n",
    "Análisis del output del caso 18NR-2."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>functional_class</th>\n",
       "      <th>gene</th>\n",
       "      <th>variant_score</th>\n",
       "      <th>pheno_score</th>\n",
       "      <th>gene_variant_score</th>\n",
       "      <th>gene_combined_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr19</td>\n",
       "      <td>6495816</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>TUBB4A</td>\n",
       "      <td>0.999295</td>\n",
       "      <td>0.765441</td>\n",
       "      <td>0.999295</td>\n",
       "      <td>0.979074</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr9</td>\n",
       "      <td>140777306</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>CACNA1B</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.737432</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.972373</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr9</td>\n",
       "      <td>141016163</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>CACNA1B</td>\n",
       "      <td>0.994195</td>\n",
       "      <td>0.737432</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.972373</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr22</td>\n",
       "      <td>24167605</td>\n",
       "      <td>splice_region_variant</td>\n",
       "      <td>SMARCB1</td>\n",
       "      <td>0.900000</td>\n",
       "      <td>0.817198</td>\n",
       "      <td>0.900000</td>\n",
       "      <td>0.969871</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr6</td>\n",
       "      <td>170871051</td>\n",
       "      <td>frameshift_truncation</td>\n",
       "      <td>TBP</td>\n",
       "      <td>0.950000</td>\n",
       "      <td>0.713726</td>\n",
       "      <td>0.950000</td>\n",
       "      <td>0.945593</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr6</td>\n",
       "      <td>170871054</td>\n",
       "      <td>frameshift_truncation</td>\n",
       "      <td>TBP</td>\n",
       "      <td>0.950000</td>\n",
       "      <td>0.713726</td>\n",
       "      <td>0.950000</td>\n",
       "      <td>0.945593</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr11</td>\n",
       "      <td>26621221</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>ANO3</td>\n",
       "      <td>0.999983</td>\n",
       "      <td>0.656656</td>\n",
       "      <td>0.999983</td>\n",
       "      <td>0.938263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr19</td>\n",
       "      <td>50364706</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>PNKP</td>\n",
       "      <td>0.998984</td>\n",
       "      <td>0.655126</td>\n",
       "      <td>0.998984</td>\n",
       "      <td>0.936794</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr6</td>\n",
       "      <td>33419613</td>\n",
       "      <td>frameshift_truncation</td>\n",
       "      <td>SYNGAP1</td>\n",
       "      <td>0.950000</td>\n",
       "      <td>0.692028</td>\n",
       "      <td>0.950000</td>\n",
       "      <td>0.932754</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr6</td>\n",
       "      <td>33419615</td>\n",
       "      <td>frameshift_elongation</td>\n",
       "      <td>SYNGAP1</td>\n",
       "      <td>0.950000</td>\n",
       "      <td>0.692028</td>\n",
       "      <td>0.950000</td>\n",
       "      <td>0.932754</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      chr        pos       functional_class     gene  variant_score  \\\n",
       "0   chr19    6495816       missense_variant   TUBB4A       0.999295   \n",
       "1    chr9  140777306       missense_variant  CACNA1B       1.000000   \n",
       "3    chr9  141016163       missense_variant  CACNA1B       0.994195   \n",
       "4   chr22   24167605  splice_region_variant  SMARCB1       0.900000   \n",
       "5    chr6  170871051  frameshift_truncation      TBP       0.950000   \n",
       "6    chr6  170871054  frameshift_truncation      TBP       0.950000   \n",
       "8   chr11   26621221       missense_variant     ANO3       0.999983   \n",
       "10  chr19   50364706       missense_variant     PNKP       0.998984   \n",
       "11   chr6   33419613  frameshift_truncation  SYNGAP1       0.950000   \n",
       "12   chr6   33419615  frameshift_elongation  SYNGAP1       0.950000   \n",
       "\n",
       "    pheno_score  gene_variant_score  gene_combined_score  \n",
       "0      0.765441            0.999295             0.979074  \n",
       "1      0.737432            1.000000             0.972373  \n",
       "3      0.737432            1.000000             0.972373  \n",
       "4      0.817198            0.900000             0.969871  \n",
       "5      0.713726            0.950000             0.945593  \n",
       "6      0.713726            0.950000             0.945593  \n",
       "8      0.656656            0.999983             0.938263  \n",
       "10     0.655126            0.998984             0.936794  \n",
       "11     0.692028            0.950000             0.932754  \n",
       "12     0.692028            0.950000             0.932754  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "caso9 = 'path_to_18NR-2_output.variants.tsv'\n",
    "nimgenetics.filterVariants(data=caso9)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Posición de la variante genética indicada por los analistas en la lista de genes generada por exomiser."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>functional_class</th>\n",
       "      <th>gene</th>\n",
       "      <th>variant_score</th>\n",
       "      <th>pheno_score</th>\n",
       "      <th>gene_variant_score</th>\n",
       "      <th>gene_combined_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr11</td>\n",
       "      <td>26581382</td>\n",
       "      <td>coding_transcript_intron_variant</td>\n",
       "      <td>ANO3</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.656656</td>\n",
       "      <td>0.999983</td>\n",
       "      <td>0.938263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr11</td>\n",
       "      <td>26621221</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>ANO3</td>\n",
       "      <td>0.999983</td>\n",
       "      <td>0.656656</td>\n",
       "      <td>0.999983</td>\n",
       "      <td>0.938263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr11</td>\n",
       "      <td>26645955</td>\n",
       "      <td>coding_transcript_intron_variant</td>\n",
       "      <td>ANO3</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.656656</td>\n",
       "      <td>0.999983</td>\n",
       "      <td>0.938263</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     chr       pos                  functional_class  gene  variant_score  \\\n",
       "7  chr11  26581382  coding_transcript_intron_variant  ANO3       0.000000   \n",
       "8  chr11  26621221                  missense_variant  ANO3       0.999983   \n",
       "9  chr11  26645955  coding_transcript_intron_variant  ANO3       0.000000   \n",
       "\n",
       "   pheno_score  gene_variant_score  gene_combined_score  \n",
       "7     0.656656            0.999983             0.938263  \n",
       "8     0.656656            0.999983             0.938263  \n",
       "9     0.656656            0.999983             0.938263  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nimgenetics.locVariant(data=caso9, gen_sel='ANO3')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Caso 18NR-3\n",
    "\n",
    "Análisis del output del caso 18NR-3."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>functional_class</th>\n",
       "      <th>gene</th>\n",
       "      <th>variant_score</th>\n",
       "      <th>pheno_score</th>\n",
       "      <th>gene_variant_score</th>\n",
       "      <th>gene_combined_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr4</td>\n",
       "      <td>1806119</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>FGFR3</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.998149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr7</td>\n",
       "      <td>27140845</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>HOXA2</td>\n",
       "      <td>0.999999</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.999999</td>\n",
       "      <td>0.998149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>185</th>\n",
       "      <td>chr1</td>\n",
       "      <td>235894211</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>LYST</td>\n",
       "      <td>0.997261</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.997261</td>\n",
       "      <td>0.998102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>187</th>\n",
       "      <td>chr6</td>\n",
       "      <td>31239006</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>HLA-B</td>\n",
       "      <td>0.997000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.997000</td>\n",
       "      <td>0.998097</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>188</th>\n",
       "      <td>chr6</td>\n",
       "      <td>31239111</td>\n",
       "      <td>frameshift_elongation</td>\n",
       "      <td>HLA-B</td>\n",
       "      <td>0.950000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.997000</td>\n",
       "      <td>0.998097</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>189</th>\n",
       "      <td>chr6</td>\n",
       "      <td>31239113</td>\n",
       "      <td>frameshift_truncation</td>\n",
       "      <td>HLA-B</td>\n",
       "      <td>0.950000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.997000</td>\n",
       "      <td>0.998097</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>190</th>\n",
       "      <td>chr6</td>\n",
       "      <td>31239133</td>\n",
       "      <td>splice_region_variant</td>\n",
       "      <td>HLA-B</td>\n",
       "      <td>0.776014</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.997000</td>\n",
       "      <td>0.998097</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191</th>\n",
       "      <td>chr6</td>\n",
       "      <td>31323115</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>HLA-B</td>\n",
       "      <td>0.723000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.997000</td>\n",
       "      <td>0.998097</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193</th>\n",
       "      <td>chr6</td>\n",
       "      <td>31324100</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>HLA-B</td>\n",
       "      <td>0.488000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.997000</td>\n",
       "      <td>0.998097</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>195</th>\n",
       "      <td>chr4</td>\n",
       "      <td>151829506</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>LRBA</td>\n",
       "      <td>0.981000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.981000</td>\n",
       "      <td>0.997797</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      chr        pos       functional_class   gene  variant_score  \\\n",
       "1    chr4    1806119       missense_variant  FGFR3       1.000000   \n",
       "3    chr7   27140845       missense_variant  HOXA2       0.999999   \n",
       "185  chr1  235894211       missense_variant   LYST       0.997261   \n",
       "187  chr6   31239006       missense_variant  HLA-B       0.997000   \n",
       "188  chr6   31239111  frameshift_elongation  HLA-B       0.950000   \n",
       "189  chr6   31239113  frameshift_truncation  HLA-B       0.950000   \n",
       "190  chr6   31239133  splice_region_variant  HLA-B       0.776014   \n",
       "191  chr6   31323115       missense_variant  HLA-B       0.723000   \n",
       "193  chr6   31324100       missense_variant  HLA-B       0.488000   \n",
       "195  chr4  151829506       missense_variant   LRBA       0.981000   \n",
       "\n",
       "     pheno_score  gene_variant_score  gene_combined_score  \n",
       "1            1.0            1.000000             0.998149  \n",
       "3            1.0            0.999999             0.998149  \n",
       "185          1.0            0.997261             0.998102  \n",
       "187          1.0            0.997000             0.998097  \n",
       "188          1.0            0.997000             0.998097  \n",
       "189          1.0            0.997000             0.998097  \n",
       "190          1.0            0.997000             0.998097  \n",
       "191          1.0            0.997000             0.998097  \n",
       "193          1.0            0.997000             0.998097  \n",
       "195          1.0            0.981000             0.997797  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "caso10 = 'path_to_18NR-3_output.variants.tsv'\n",
    "nimgenetics.filterVariants(data=caso10)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Posición de la variante genética indicada por los analistas en la lista de genes generada por exomiser."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>functional_class</th>\n",
       "      <th>gene</th>\n",
       "      <th>variant_score</th>\n",
       "      <th>pheno_score</th>\n",
       "      <th>gene_variant_score</th>\n",
       "      <th>gene_combined_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr4</td>\n",
       "      <td>1800914</td>\n",
       "      <td>coding_transcript_intron_variant</td>\n",
       "      <td>FGFR3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.998149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr4</td>\n",
       "      <td>1806119</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>FGFR3</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.998149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr4</td>\n",
       "      <td>1808495</td>\n",
       "      <td>coding_transcript_intron_variant</td>\n",
       "      <td>FGFR3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.998149</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chr      pos                  functional_class   gene  variant_score  \\\n",
       "0  chr4  1800914  coding_transcript_intron_variant  FGFR3            0.0   \n",
       "1  chr4  1806119                  missense_variant  FGFR3            1.0   \n",
       "2  chr4  1808495  coding_transcript_intron_variant  FGFR3            0.0   \n",
       "\n",
       "   pheno_score  gene_variant_score  gene_combined_score  \n",
       "0          1.0                 1.0             0.998149  \n",
       "1          1.0                 1.0             0.998149  \n",
       "2          1.0                 1.0             0.998149  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nimgenetics.locVariant(data=caso10, gen_sel='FGFR3')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
